### ISSUE TITLE
AggregatedAPIDown is constantly triggered after prometheus-operator update

### ISSUE THREAD

**Issue by den-is** [kubernetes-monitoring/kubernetes-mixin#397]
After upgrading to latest prometheus-operator (kube-prometheus ... or whatever you wish to call it)
Cluster started constantly triggering `AggregatedAPIDown` alert.
Not sure what is wrong or where else to check everything seems to be healthy :(
I'm not sure if this alert is really required or how to fix it.
Just one of the 3 apiserver returns value > 0

First issue was initiated here https://github.com/kubernetes/kube-aggregator/issues/23

Running kubernetes v1.16.8 on aws via kops

**Comment by zerkms** [comment]
The same here, on kubernetes 1.16 and 1.17.

Can we please disable that rule until it becomes more reliable.

**Comment by den-is** [comment]
For now, I have just removed the definition of that alert after install.
Indeed Kubernetes 1.18 has not the same issue. But i'm not sure that majority of cluster will have 1.18 anytime soon.

**Comment by povilasv** [comment]
I have seen this as well, I believe Kubernetes gauge is not being updated correctly.

**Comment by povilasv** [comment]
FYI you can write jsonnet:

```
+ {
  prometheusAlerts+:: {
    groups:
      std.map(
        function(group)
          if group.name == 'kubernetes-system-apiserver' then
            group {
              rules: std.filter(
                function(rule)
                  rule.alert != 'AggregatedAPIDown',
                group.rules
              ),
            }
          else
            group,
        super.groups
      ),
  },
}
```

To remove it.

**Comment by den-is** [comment]
IMHO If gauge is not updated correctly in currently widely spread Kubernetes version that means that it is not the time for that alert in massive prod.
Personally, I do not write/touch jsonnet (as many others) and install mixin-resource from helm chart https://github.com/helm/charts/tree/master/stable/prometheus-operator,
which is using another upstream project as a source https://github.com/coreos/kube-prometheus/tree/master/manifests

**Comment by TJM** [comment]
OK, I see merged PRs above that may fix this, but now I also see that the most recent version (10.x?) requires K8s 1.16.0 or higher.   SO, I am just routing the alert to blackhole for now (like Watchdog)

**Comment by akunaatrium** [comment]
I just saw the same alert on Kubernetes 1.18. using the latest version of kube-prometheus (from the master branch). Not sure yet if it starts to happen often.

Also not going to touch jsonnet code. It seems to introduce much more complexity into managing the manifests without sufficient benefits or justification. I just took the manifests folder from kube-prometheus and using Kustomize to tweak whatever needed, like adding new alerting rules or configuring Grafana. Can bypass the several additional tools besides jsonnet itself, like Go, jsonnet-bundler, gojsontoyaml, Docker, Unix based OS to execute build.sh.
